client/c/src/flora.c

```diff
@@ -33,9 +41,12 @@ static void _flora_method_call_harbor(const char* name, caps_t msg,
   flora_call_reply_write_code(reply, 0);
   flora_call_reply_end(reply);
 
+  yodaos_local_event_lock(1);
   if (_harbor_cb && type && desc) {
     _harbor_cb(name, (const char*)type, (const char*)desc);
   }
+  yodaos_local_event_lock(0);
+  usleep(100);
```

Comment:
Is the usleep necessary here?

Reply:
It’s necessary. Add sleep to schedule another thread. In extreme circumstances, the ping thread cannot exec and do not send alive msg. Thus, runtime kill the process.